centreon/src/App/Security/Infrastructure/Security/TokenAuthenticator.php

```
@@ -0,0 +1,135 @@
+    public function authenticate(Request $request): Passport
+    {
+        if (! $token = $request->headers->get('X-AUTH-TOKEN')) {
+            throw new AuthenticationCredentialsNotFoundException();
+        }
+
+        return new SelfValidatingPassport(
+            new UserBadge($token, $this->getCredentialUser(...)),
+        );
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): null
```

PHPMD (2.15.0)

Avoid unused parameters such as '$firewallName'. (undefined)

(UnusedFormalParameter)